tests/validator_test.go

```diff
@@ -408,7 +409,21 @@ var _ = Describe("Template validator", func() {
 		It("[test_id:2960] Negative test - Create a VM with machine type violation", func() {
 			template = TemplateWithRules()
 			Expect(apiClient.Create(ctx, template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+			// set value unfulfilling validation
+			vmi = addDomainResourcesToVMI(vmi, 2, "test", "128M")
+			vm = NewVirtualMachine(vmi)
+			vm.ObjectMeta.Annotations = map[string]string{
+				TemplateNameAnnotation:      template.Name,
+				TemplateNamespaceAnnotation: template.Namespace,
+			}
+			eventuallyFailToCreateVm(vm)
+			Expect(errors.IsInvalid(apiClient.Create(ctx, vm))).To(BeTrue(), "Should match error type because of unfulfilled validations")
+		})
 
+		It("[test_id:7060]vm rejected metrics increses by one when vm is rejected", func() {
+			defer expectTotalRejectedCountToIncBy(1)()
```

Comment:
I think it is problematic to call `Expect()` in a function that is executed using `defer`.
If this test case fails, for example on line 426, the `defer` function will still be executed.

Can you inline this function like:
```go
It("[test_id:7060]vm rejected metrics increses by one when vm is rejected", func() {
    rejectedBefore := totalRejectedVmsMetricsValue()

    // ... the rest of test case

    Expect(totalRejectedVmsMetricsValue() - rejectedBefore).To(Equal(1))
}
```

Reply:
hmm, that is what I had in the beginning and I found the deffer thing nicer - one line in the beginning that explains the essence of the test right away. is it really problematic if this deffer runs in any case? We will just get another failure message right? And it will be true...

Reply:
I'm not sure how golang handles `panic` inside `defer` function that is called while panicking.
And the function `totalRejectedVmsMetricsValue()` is not fast, it calls API. So it would be better to not call it if not necessary.

Reply:
ok